pypeit/core/extract.py

```diff
@@ -206,6 +206,7 @@ def extract_boxcar(sciimg, ivar, mask, waveimg, skyimg, rn2_img, box_radius, spe
         spec.trace_spec = spec_vec
 
     # TODO This makes no sense for difference imaging? Not sure we need NIVAR anyway
+    # TODO: This needs to be corrected for new noise model approach.
```

Comment:
Is that to be done in this PR?

Reply:
I'm happy to do that in this PR, but I need input from @jhennawi .

Reply:
Thanks for this explanation, @jhennawi .  Can you also explain the thinking behind `OPT_COUNTS_RN`?  I've been updating the calculations in the extraction outputs, and I wanted to know if this should similarly be updated to include more than just the readnoise terms.

Reply:
I've updated the `extract_optimal` and `extract_boxcar` to use the new variance calculations.  This led to a datamodel change and a change to the input arguments.  These functions no longer require the RN variance image, but take the "base-level" variance image as an optional argument.  The "base-level" variance image is calculated by `pypeit.core.procimg.base_variance`, and it includes the readnoise variance as one of its factors.  Also, the `OPT_COUNTS_RN` component of the datamodel has been renamed to `OPT_COUNTS_SIG_DET`; similarly, `BOX_COUNTS_RN` -> `BOX_COUNTS_SIG_DET`.